packages/localize/src/number/forceAddGroupSeparators.js

```diff
@@ -7,38 +7,44 @@
  */
 export function forceAddGroupSeparators(formattedParts, groupSeparator) {
   let concatArray = [];
-  if (formattedParts[0].type === 'integer') {
-    const getInteger = formattedParts.splice(0, 1);
-    const numberOfDigits = getInteger[0].value.length;
-    const mod3 = numberOfDigits % 3;
-    const groups = Math.floor(numberOfDigits / 3);
-    const numberArray = [];
-    let numberOfGroups = 0;
-    let numberPart = '';
-    let firstGroup = false;
-    // Loop through the  integer
-    for (let i = 0; i < numberOfDigits; i += 1) {
-      numberPart += getInteger[0].value[i];
-      // Create first grouping which is < 3
-      if (numberPart.length === mod3 && firstGroup === false) {
-        numberArray.push({ type: 'integer', value: numberPart });
-        if (numberOfDigits > 3) {
-          numberArray.push({ type: 'group', value: groupSeparator });
-        }
-        numberPart = '';
-        firstGroup = true;
-        // Create groupings of 3
-      } else if (numberPart.length === 3 && i < numberOfDigits - 1) {
-        numberOfGroups += 1;
-        numberArray.push({ type: 'integer', value: numberPart });
-        if (numberOfGroups !== groups) {
-          numberArray.push({ type: 'group', value: groupSeparator });
-        }
-        numberPart = '';
+  let firstPart;
+  let integerPart;
+
+  for (let i = 0; i < formattedParts.length; i += 1) {
```

Comment:
for (let i = 0; i < formattedParts.length; i += 1) {
    if (formattedParts[i].type === 'integer') {
      firstPart = formattedParts.splice(0, i);
      integerPart = formattedParts.splice(0, 1);
    } else {
      return concatArray;
    }
  }

Reply:
This suggestion does not work, but i found another way.